torch/utils/bottleneck/__main__.py

```diff
@@ -0,0 +1,231 @@
+    print('cProfile output{}'.format(env_str))
+    print_line()
+    cprofile_stats = pstats.Stats(prof).sort_stats(sortby)
+    cprofile_stats.print_stats(topk)
+
+
+def run_autograd_prof(code, globs, gpu_device=0):
+    set_env_variables(gpu_device)
+    env_str = get_env_variables_description(gpu_device)
+    print('Running your script with the autograd profiler{}...'.format(env_str))
+    with profiler.profile() as prof:
+        exec(code, globs, None)
+    return prof
+
+
+def print_autograd_prof_summary(prof, gpu_device=0, sortby='cpu_time', topk=15):
+    valid_sortby = ['cpu_time', 'cuda_time', 'cpu_time_total', 'cuda_time_total', 'count']
+    if sortby not in valid_sortby:
+        warn = ('WARNING: invalid sorting option for autograd profiler results: {}\n'
+                'Expected `cpu_time`, `cpu_time_total`, or `count`. '
+                'Defaulting to `cpu_time`.')
+        print(warn.format(autograd_prof_sortby))
+        sortby = 'cpu_time'
+
+    print_line()
+    env_str = get_env_variables_description(gpu_device, '\n')
+    print('autograd profiler output{}'.format(env_str))
+    print_line()
+    print('\ttop {} events sorted by {}'.format(topk, sortby))
+    ex = ('    Note that because CUDA_LAUNCH_BLOCKING=1 is set, the reported CPU time\n'
+          '    includes the CUDA time. Ignore the empty CUDA time column here.\n')
+    if torch.cuda.is_available():
+        print('')
+        print(ex)
+
+    sorted_events = sorted(prof.function_events,
+                           key=lambda x: getattr(x, sortby), reverse=True)
+    topk_events = sorted_events[:topk]
+    print(torch.autograd.profiler.build_table(topk_events))
+
+
+descript = ('`bottleneck` is a tool that can be used as an initial step for debugging '
+            'bottlenecks in your program.\n\n'
+            'It summarizes runs of your script with the Python profiler '
+            'and PyTorch\'s autograd profiler. For ease of use and intepretability of '
+            'results, `bottleneck` runs multi-GPU code on only one GPU device. '
+            'Because your script will be profiled, please ensure that it exits '
+            'in a finite amount of time. \n\n'
+            'For more complicated uses of the profilers (like in a multi-GPU case), '
+            'please see https://docs.python.org/3/library/profile.html '
+            'and http://pytorch.org/docs/master/autograd.html#profiler '
+            'for more information. \n')
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description=descript)
+    parser.add_argument('scriptfile', type=str,
+                        help='Path to the script to be run. '
+                        'Usually run with `python path/to/script`.')
+    parser.add_argument('--gpu', dest='gpu_device', type=int, default=0,
+                        help='If applicable, which GPU device to run '
+                        'on. Default: 0.')
+    return parser.parse_args()
+    return parser.parse_args()
```

Comment:
Noticed this line was repeated. Cool utility! 😄

Reply:
Thanks for catching that!